core/src/main/java/io/paradoxical/cassieq/admin/resources/api/v1/PermissionsResource.java

```diff
@@ -85,17 +88,26 @@ public Response generateAuthUrl(@NotNull GetAuthQueryParamsRequest request) thro
             return Response.status(Response.Status.NOT_FOUND).build();
         }
 
-        final SignedUrlParameterGenerator signedUrlParameterGenerator = new SignedUrlParameterGenerator(request.getAccountName(), authorizationLevels);
+        final SignedUrlSignatureGenerator signedUrlParameterGenerator =
+                new SignedUrlSignatureGenerator(
+                        request.getAccountName(),
+                        authorizationLevels,
+                        request.getStartTime(),
+                        request.getEndTime());
 
         final AccountKey key = keys.get(request.getKeyName());
 
         final String computedSignature = signedUrlParameterGenerator.computeSignature(MacProviders.HmacSha256(key));
 
-        final String queryParam = SignedUrlParameterNames.builder()
+        final String queryParam = SignedUrlParameterNames.queryBuilder()
                                                          .auth(authorizationLevels)
-                                                         .sig(computedSignature)
+                                                         .startTime(request.getStartTime())
+                                                         .endTime(request.getEndTime())
+                                                         .sig(computedSignature) // always make this last (for prettiness)
```

Comment:
Good point. its not a HUGE deal, but order matters now because i made it a linked hash map